superset-frontend/src/components/Menu/SubMenu.tsx

```diff
@@ -72,43 +73,56 @@ interface Props {
 
 interface State {
   selectedMenu: string;
+  isModalOpen: boolean;
 }
 
 class SubMenu extends React.PureComponent<Props, State> {
   state: State = {
     selectedMenu: this.props.childs[0] && this.props.childs[0].label,
+    isModalOpen: false,
   };
 
   handleClick = (item: string) => () => {
     this.setState({ selectedMenu: item });
   };
 
+  openModal = () => {
+    this.setState({ isModalOpen: true });
+  };
+
+  closeModal = () => {
```

Comment:
should we call this `hideModal` to be consistent with the `onHide` attribute? or rename the modal's prop to `onClose`. Either one seems fine, although i slightly prefer updating the prop to `onClose` and keeping this as is